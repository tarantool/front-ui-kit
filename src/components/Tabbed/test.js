import React from 'react';
import renderer from 'react-test-renderer';
import { Tabbed } from './index';

describe('Tabbed', () => {
  it('Tabbed renders correctly', () => {
    const tree = renderer.create(
      <Tabbed
        tabs={[
          {
            label: 'Create Replica Set',
            content: 'Tab content'
          },
          {
            label: 'Another tab',
            content: 'Another tab content'
          },
          {
            label: 'Third tab',
            content: 'Third tab content'
          }
        ]}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Tabbed small renders correctly', () => {
    const tree = renderer.create(
      <Tabbed
        size="small"
        tabs={[
          {
            label: 'Create Replica Set',
            content: 'Tab content'
          },
          {
            label: 'Another tab',
            content: 'Another tab content'
          },
          {
            label: 'Third tab',
            content: 'Third tab content'
          }
        ]}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

