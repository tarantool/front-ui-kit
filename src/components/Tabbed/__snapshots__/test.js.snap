// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabbed Tabbed renders correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.emotion-1 {
  border-bottom: solid 1px #d9dadd;
}

.emotion-2 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #040b1d;
  -webkit-font-smoothing: antialiased;
  position: relative;
  padding: 14px 0;
  border: none;
  margin-right: 30px;
  margin-bottom: -1px;
  border-bottom: solid 2px transparent;
  color: #8e9199;
  background-color: transparent;
  outline: none;
  cursor: pointer;
  color: #040b1d;
  border-bottom-color: #040b1d;
}

.emotion-2:hover,
.emotion-2:focus {
  z-index: 1;
  color: #5c606c;
}

.emotion-2:last-child {
  margin-right: 0;
}

.emotion-3 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #040b1d;
  -webkit-font-smoothing: antialiased;
  position: relative;
  padding: 14px 0;
  border: none;
  margin-right: 30px;
  margin-bottom: -1px;
  border-bottom: solid 2px transparent;
  color: #8e9199;
  background-color: transparent;
  outline: none;
  cursor: pointer;
}

.emotion-3:hover,
.emotion-3:focus {
  z-index: 1;
  color: #5c606c;
}

.emotion-3:last-child {
  margin-right: 0;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      Create Replica Set
    </button>
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      Another tab
    </button>
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      Third tab
    </button>
  </div>
  Tab content
</div>
`;

exports[`Tabbed Tabbed small renders correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  background-color: #f4f4f4;
  border-radius: 4px;
}

.emotion-2 {
  position: relative;
  cursor: pointer;
  border: none;
  font-size: 14px;
  color: #040b1d;
  width: 100%;
  margin-left: 8px;
  margin-right: 8px;
  padding: 9px 20px;
  border-radius: 4px;
  background-color: #f4f4f4;
  background-color: #fff!important;
  box-shadow: 0px 2px 2px rgba(4, 11, 29, 0.05);
}

.emotion-2:focus {
  background-color: #f4f4f4;
}

.emotion-2:hover {
  z-index: 1;
  background-color: #fff;
}

.emotion-3 {
  position: relative;
  cursor: pointer;
  border: none;
  font-size: 14px;
  color: #040b1d;
  width: 100%;
  margin-left: 8px;
  margin-right: 8px;
  padding: 9px 20px;
  border-radius: 4px;
  background-color: #f4f4f4;
}

.emotion-3:focus {
  background-color: #f4f4f4;
}

.emotion-3:hover {
  z-index: 1;
  background-color: #fff;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-2"
      onClick={[Function]}
      type="button"
    >
      Create Replica Set
    </button>
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      Another tab
    </button>
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      Third tab
    </button>
  </div>
  Tab content
</div>
`;
